drivers/sensor/ltrf216a/ltrf216a.c

```diff
@@ -0,0 +1,233 @@
+
+#define LTRF216A_ALS_DATA_STATUS 8
+
+/* Part Number ID 7:4 0b1011 */
+#define LTRF216A_PART_ID_PARTNUMBERID_VALUE 0xB
+#define LTRF216A_PART_ID_PARTNUMBERID_MASK  0xF0
+#define LTRF216A_PART_ID_PARTNUMBERID_SHIFT 4
+
+/* Revision ID 3:0 0b0001 */
+#define LTRF216A_PART_ID_REVISIONID_VALUE 0x1
+#define LTRF216A_PART_ID_REVISIONID_MASK  0x0F
+
+#define LTRF216A_MAIN_CTRL        0x00
+#define LTRF216A_ALS_MEAS_RES     0x04
+#define LTRF216A_ALS_GAIN         0x05
+#define LTRF216A_PART_ID          0x06
+#define LTRF216A_MAIN_STATUS      0x07
+#define LTRF216A_ALS_CLEAR_DATA_0 0x0a
+#define LTRF216A_ALS_CLEAR_DATA_1 0x0b
+#define LTRF216A_ALS_CLEAR_DATA_2 0x0c
+#define LTRF216A_ALS_DATA_0       0x0d
+#define LTRF216A_ALS_DATA_1       0x0e
+#define LTRF216A_ALS_DATA_2       0x0f
+#define LTRF216A_INT_CFG          0x19
+#define LTRF216A_INT_PST          0x1a
+#define LTRF216A_ALS_THRES_UP_0   0x21
+#define LTRF216A_ALS_THRES_UP_1   0x22
+#define LTRF216A_ALS_THRES_UP_2   0x23
+#define LTRF216A_ALS_THRES_LOW_0  0x24
+#define LTRF216A_ALS_THRES_LOW_1  0x25
+#define LTRF216A_ALS_THRES_LOW_2  0x26
+
+#define LTRF216A_ALS_READ_DATA_DELAY_MS 20
+
+#define LTRF216A_WIN_FAC 1
+
+#define LTRF216A_NUMBER_DATA_REGISTERS 6
+
+static int ltrf216a_reg_read(const struct device *dev, uint8_t reg, uint8_t *val)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	if (i2c_burst_read_dt(&config->i2c, reg, val, 1) != 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int ltrf216a_reg_read_multiple(const struct device *dev, uint8_t reg, uint8_t *val,
+				      uint32_t size)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	if (i2c_burst_read_dt(&config->i2c, reg, val, size) != 0) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int ltrf216a_reg_write(const struct device *dev, uint8_t reg, uint8_t val)
+{
+	const struct ltrf216a_config *config = dev->config;
+
+	return i2c_write_dt(&config->i2c, &val, 1);
+}
+
+static int ltrf216a_enable(const struct device *dev)
+{
+	if (ltrf216a_reg_write(dev, LTRF216A_MAIN_CTRL, LTRF216A_ALS_ENABLE_MASK) != 0) {
+		LOG_ERR("Failed to enable sensor");
+		return -EIO;
+	}
+	return 0;
+}
+
+static int ltrf216a_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct ltrf216a_data *drv_data = dev->data;
+	uint8_t status;
+	uint8_t value[LTRF216A_NUMBER_DATA_REGISTERS];
+	int result;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT);
+
+	drv_data->sample[0] = 0U;
+	drv_data->sample[1] = 0U;
+	drv_data->sample[2] = 0U;
+
+	result = ltrf216a_enable(dev);
+	if (result != 0) {
+		LOG_ERR("ltfr216a: enable failed");
+		return result;
+	}
+
+	while ((result = ltrf216a_reg_read(dev, LTRF216A_MAIN_STATUS, &status)) == 0) {
+		if ((status & LTRF216A_ALS_DATA_STATUS) != 0) {
+			break;
+		}
+
+		k_sleep(K_MSEC(LTRF216A_ALS_READ_DATA_DELAY_MS));
+	}
+
+	if (ltrf216a_reg_read_multiple(dev, LTRF216A_ALS_CLEAR_DATA_0, value,
+				       LTRF216A_NUMBER_DATA_REGISTERS) != 0) {
```

Comment:
Could you pass drv_data->sample here instead of using a temporary variable?

Reply:
Need to check on the device, whether reading needs to start at LTRF216A_ALS_CLEAR_DATA_0 with 6 bytes or reading from LTRF216A_ALS_DATA_0 with 3 bytes works as well.

Might take until end of next week.

Reply:
Is the clear channel likely to be needed in the future?

Reply:
CLEAR_DATA is not necessarily required. We are only using the ALS channel which has a different Spectral Response (see Figure 4.1 of the datasheet).

applied the recommendation in the latest force push.